packages/core/strapi/src/node/create-build-context.ts

```diff
@@ -79,15 +80,25 @@ const createBuildContext = async <TOptions extends BaseOptions>({
     });
 
   const serverUrl = strapiInstance.config.get<string>('server.url');
-  const adminPath = strapiInstance.config.get<string>('admin.path');
+  const serverAbsoluteUrl = strapiInstance.config.get<string>('server.absoluteUrl');
+  const adminUrl = strapiInstance.config.get<string>('admin.url');
+  const adminAbsoluteUrl = strapiInstance.config.get<string>('admin.absoluteUrl');
+
+  const sameOrigin = new URL(adminAbsoluteUrl).origin === new URL(serverAbsoluteUrl).origin;
```

Comment:
Maybe consider adding a quick explanation of what we're testing.

e.g.

```typescript
// Check if both the server and admin URLs share the same origin
// Example: Comparing origins 'http://localhost:1337/admin' and 'http://localhost:1337/cms' will return true
```

Reply:
this doesnt seem to work, if the admin config URL is set to `/` and the server URL is set to a backend URL. this still results in true and hence admin panel is built with relative path. essentially deployment on different server doesn't work.

It will be good to utilise the serverAdminPanel config option itself to determine the backend URL.